sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -123,26 +123,41 @@ case class ConcatWs(children: Seq[Expression])
       val idxVararg = ctx.freshName("idxInVararg")
 
       val evals = children.map(_.genCode(ctx))
+      val supportSplitExpressions = ctx.supportSplitExpressionsWithCurrentInputs
```

Comment:
We may be able to employ some threshold of the number of children or fallback logic if we want to also exclude the case where the number of blocks for all three parts are all 1, but that's going to be complicated a bit, so would like to hear others' voices before applying.